lib/Backend/PageAllocatorPool.cpp

```diff
@@ -37,8 +38,10 @@ void PageAllocatorPool::Shutdown()
     AutoCriticalSection autoCS(&cs);
     if (Instance)
     {
-        CloseHandle(Instance->idleCleanupTimer);
-        Instance->RemoveAll();
+        if (Instance->idleCleanupTimer)
+        {
+            CloseHandle(Instance->idleCleanupTimer);
```

Comment:
What is the guarantee that it will not re-enter again (in future changes) ?

Other practice you can do that 
tempInstance = Instance
Instance = nullptr;
HeapDelete(tempInstance).
that way if the Shutdown calls itself - we will be safe.



---
In reply to: [86002065](https://github.com/Microsoft/ChakraCore/pull/1892#discussion_r86002065) [](ancestors = 86002065)